services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantService.java

```diff
@@ -0,0 +1,528 @@
+                .compose(ok -> processUpdateTenant(tenantId, tenantObj, resourceVersion, span))
+                .recover(error -> Future.succeededFuture(MongoDbDeviceRegistryUtils.mapErrorToResult(error, span)));
+    }
+
+    private Future<OperationResult<Void>> processUpdateTenant(final String tenantId, final Tenant newTenant,
+            final Optional<String> resourceVersion, final Span span) {
+        Objects.requireNonNull(tenantId);
+        Objects.requireNonNull(newTenant);
+        Objects.requireNonNull(resourceVersion);
+        Objects.requireNonNull(span);
+
+        final JsonObject updateTenantQuery = resourceVersion
+                .map(version -> new MongoDbDocumentBuilder().withVersion(version))
+                .orElse(new MongoDbDocumentBuilder())
+                .withTenantId(tenantId)
+                .document();
+
+        final Promise<JsonObject> updateTenantPromise = Promise.promise();
+        final TenantDto newTenantDto = new TenantDto(tenantId, newTenant,
+                new Versioned<>(newTenant).getVersion());
+        mongoClient.findOneAndReplaceWithOptions(config.getCollectionName(), updateTenantQuery,
+                JsonObject.mapFrom(newTenantDto), new FindOptions(), new UpdateOptions().setReturningNewDocument(true),
+                updateTenantPromise);
+
+        return updateTenantPromise.future()
+                .compose(updateResult -> Optional.ofNullable(updateResult)
+                        .map(updated -> {
+                            span.log(String.format("successfully updated tenant [%s]", tenantId));
+                            return Future.<OperationResult<Void>> succeededFuture(OperationResult.ok(
+                                    HttpURLConnection.HTTP_NO_CONTENT,
+                                    null,
+                                    Optional.empty(),
+                                    Optional.of(updateResult.getString(MongoDbDeviceRegistryUtils.FIELD_VERSION))));
+                        })
+                        .orElse(MongoDbDeviceRegistryUtils.checkForVersionMismatchAndFail(
+                                tenantId, resourceVersion, findTenant(tenantId))))
+                .recover((error) -> {
+                    if (error instanceof MongoCommandException) {
+                        logDuplicationError(((MongoCommandException) error).getErrorMessage(), tenantId, span);
+                        return Future.succeededFuture(OperationResult.empty(HttpURLConnection.HTTP_CONFLICT));
+                    } else if (error instanceof ServiceInvocationException) {
+                        return Future.failedFuture(((ServiceInvocationException) error));
+                    }
+                    return Future.failedFuture(new ServerErrorException(HttpURLConnection.HTTP_INTERNAL_ERROR,
+                            String.format("Error modifying tenant [%s].", tenantId)));
+                });
+    }
+
+//
```

Comment:
I guess its by mistake and pls remove it.

Reply:
fixed